wordpress-mobile/WordPress-Android
# Jetpack: Hide Restore Option for Unavailable Rewind State

Fixes #14945
Related to: [WordPress-FluxC-Android#2044](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2044)

This PR is using its related PR, which provides the new `Reason` enum, to utilise that with the `GetRestoreStatusUseCase.kt` use case in order to emit the new `RestoreRequestState.Multisite` state to hide the restore option for an unavailable rewind state on multisites.

There are two screens that are affected by this change, those are the:
- `Activity Log` (or `Backup`) screen and
- `Event Details` screen

Site | Activity Log | Event Details
-----|-----|-----
Not Multisite | <img width="250" height="500" alt="activity_log_not_multisite" src="https://user-images.githubusercontent.com/9729923/124434846-0a656600-dd7d-11eb-9d1b-cc4f042f0131.png"> | <img width="250" height="500" alt="event_details_not_multisite" src="https://user-images.githubusercontent.com/9729923/124434979-2e28ac00-dd7d-11eb-83eb-a1532ed3bba4.png">
Multisite | <img width="250" height="500" alt="activity_log_multisite" src="https://user-images.githubusercontent.com/9729923/124435010-37197d80-dd7d-11eb-84d7-02a49e0e8e70.png"> | <img width="250" height="500" alt="event_details_multisite" src="https://user-images.githubusercontent.com/9729923/124435047-400a4f00-dd7d-11eb-875f-5c6487b4d944.png">

To test:
1) Open the `Jurassic Ninja` site: https://jurassic.ninja/specialops/
2) Pick a Jetpack plan (use `Jetpack Complete`).
3) Launch a `Jetpack Mutlisite on subdomains` site.
4) Connect to this site and open the app.
5) Go to: `My Site` -> `Activity Log` (or `Backup`)
6) Tap the menu option (`3 dots`) on an activity log item
7) Verify that the `Restore to this point` option is not available.
8) Tap on the activity log item itself to enter the `Event Details` screen.
9) Verify that the `RESTORE` button is not available.
10) Verify that a mutlisite message at the bottom is displayed, which informs the users about downloadable backups.
11) Tap on the `visit our documentation page` link within the multisite message itself.
12) Verify that you get redirected to: https://jetpack.com/support/backup/

#### Note:
Due to the nature of the `Activity Log` (or `Backup`) and `Event Details` screens, being too quick when tapping on the menu option on an activity log item or the activity log item itself, might not hide the `Restore to this point` option or `RESTORE` button. This is due to the fact that the call to check and hide those is happening after entering the screen and as such a potential 1/2' seconds delay exists in updating the screen(s) with the new state.

#### Merge Instructions:
- ~~Review [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2044), then merge and create a new beta tag.~~
- ~~Replace `fluxCVersion` hash with the beta tag in `build.gradle`.~~
- ~~Remove the `[Status] Not Ready for Merge` label.~~
- Merge this PR.

## Regression Notes
1. Potential unintended areas of impact

- `Activity Log` or `Backup` screen and flows.
- `Event Details` screen and flows.
- `Restore` screen and flows.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manually tested as described above in the `To test` section.

3. What automated tests I added (or what prevented me from doing so)

New tests within the below test suites:
- `GetRestoreStatusUseCaseTest.kt`
- `ActivityLogViewModelTest.kt`
- `ActivityLogDetailViewModelTest.kt`

PR submission checklist:
